### DISCUSSION TITLE
Why should I consider using this fork over the original next-pwa?

### DISCUSSION THREAD

**Discussion by SahilAggarwal2004** [DuCanhGH/next-pwa#72] [Q&A]
What are the improvements that are done in this fork of `next-pwa` other than the fact that it is well maintained unlike `next-pwa`?

**Comment by DuCanhGH** [comment]
I've done these changes to the package:
- Migrated to TS (examples are also in TS).
- Added support for `appDir` (`app/_offline/page.tsx` now also automatically enables offline fallback for document).
- Migrated from `babel` to `swc` (the original package uses `next/babel` to build fallback/custom worker).
- Moved workbox's options from general plugin options to `PluginOptions.workboxOptions`.
- Fixed grammar.
- For CJS, you will have to write `const withPWA = require("@ducanh2912/next-pwa").default()`, rather than `const withPWA = require("next-pwa")()`.
- You can add `app/~offline/page.js` to automatically enable the fallback worker.

In v8, I've also added these:
- Support `tsconfig.json` path aliases in custom workers. (8.0.0)
- Removed `subdomainPrefix` and `workboxOptions.exclude`. (8.0.0)
- Added `extendDefaultRuntimeCaching`, which allows you to extend the default runtime caching array, rather than overriding it. (8.5.0)
- Added a documentation website. (8.6.0)
- Added `aggressiveFrontEndNavCaching` - when combined with `cacheOnFrontEndNav`, it will cache `<link rel="stylesheet" href=""` and `<script src="" />` on page navigation. (8.7.0)

These are the changes in v9:
- Bumped minimum supported Next.js version to 11.0.0. (older Next versions didn't work prior to this anyways, 9.0.0)
- We fallback to `next/dist/build/swc` if `@swc/core` can't be resolved instead of requiring `@swc/core` to save disk space. (9.0.0)
- Added `swcMinify`. (9.1.0)
- Added `customWorkerPrefix`, `customWorkerSrc`, `customWorkerDest`. (9.4.0)

Which ones are improvements is up to you :)

**Comment by SahilAggarwal2004** [comment]
Woah that looks so promising. Especially the typescript support is cool! Thanks for the improvements and answering to my question :)
